storage/ios/Plugin/Storage.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+public struct StorageConfiguration {
+    var group = "CapacitorStorage"
+}
+
+@objc public class Storage: NSObject {
+    private var configuration: StorageConfiguration
+
+    public init(withConfiguration configuration: StorageConfiguration) {
+        self.configuration = configuration
+    }
+
+    @objc func get(byKey key: String) -> String? {
+        return defaults.string(forKey: applyPrefix(toKey: key))
+    }
+
+    @objc func set(_ value: String, forKey key: String) {
+        defaults.set(value, forKey: applyPrefix(toKey: key))
+    }
+
+    @objc func remove(byKey key: String) {
+        defaults.removeObject(forKey: applyPrefix(toKey: key))
+    }
+
+    @objc func removeAll() {
+        for key in keys {
+            defaults.removeObject(forKey: key)
+        }
+    }
+
+    @objc func getKeys() -> [String] {
+        return keys.map { String($0.dropFirst(prefix.count)) }
+    }
+
+    private var prefix: String {
+        return configuration.group == "NativeStorage" ? "" : configuration.group + "."
+    }
```

Comment:
This should belong to the configuration. It can be added as a private extension to keep the implementation details internal. If adopting the enum pattern, it would look like this:
```suggestion
private extension StorageConfiguration {
    var prefixString: String {
        switch self.group {
        case .native:
            return ""
        case .prefixed(let prefix):
            return prefix + "."
        }
    }
}
```